std/container/array.d

```diff
@@ -1740,6 +1745,15 @@ if (!is(immutable T == immutable bool))
     assertThrown!AssertError(array.length = 5);
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=23140
+@system unittest
+{
+    shared class C
+    {
+    }
+
+    assert(__traits(compiles, Array!C));
```

Comment:
Please just instantiate `Array!C` directly in the unittest, `__traits(compiles)` swallows the error when the test fails.

Reply:
Will do.  It should still work.